src/server/kube-store.js

```
@@ -0,0 +1,116 @@
+        this._initialized = false;
+    }
+
+    async _init() {
+        if (this._initialized) {
+            return;
+        }
+
+        try {
+            const buffer = await fs.promises.readFile(config.KUBE_SA_TOKEN_FILE);
+            this._sa_token = buffer.toString('utf8').trim();
+
+        } catch (err) {
+            throw new Error(`Could not namespace file at "${config.KUBE_SA_TOKEN_FILE}"`);
+        }
+
+        try {
+            const buffer = await fs.promises.readFile(config.KUBE_NAMESPACE_FILE);
+            this._k8s_namespace = buffer.toString('utf8').trim();
+
+        } catch (err) {
+            throw new Error(`Could not read service account token file at "${config.KUBE_NAMESPACE_FILE}"`);
+        }
+
+        this._initialized = true;
+    }
+
+    async _make_k8s_api_request(method, path, body) {
+        if (!this._initialized) {
+            throw new Error('Store is not initialized');
+        }
+
+        try {
+            const content_type = method === 'PATCH' ?
+                'application/merge-patch+json' :
+                'application/json';
+
+            const response = await make_http_request(
+                {
+                    method: method,
+                    hostname: this._service_host,
+                    port: this._service_port,
+                    path: path,
+                    rejectUnauthorized: false,
+                    headers: {
+                        'Content-Type': content_type,
+                        Accept: 'application/json',
+                        Authorization: `Bearer ${this._sa_token}`
+                    }
+                },
+                body && JSON.stringify(body),
+                'utf8'
+            );
+
+            const status_code = response.statusCode;
+            const buffer = await read_stream_join(response);
+            const res_body = JSON.parse(buffer.toString('utf8'));
+            return {
+                status_code,
```

what if the status code is not OK? maybe throw an error?

---

No, just return the status code (like I do) then let the calling code decide what to do with it. Some failures are expected

---

@dannyzaken I changed it, the make API call still returning the status code but the calling method deals with each status code according to the specific case